src/node.rs

```diff
@@ -112,10 +119,27 @@ impl Node {
 
         Some((class_name, class_version))
     }
+
+    pub fn cmp_by_id(&self, other: &Node) -> cmp::Ordering {
+        self.get_id().cmp(&other.get_id())
+    }
+
+    pub fn get_basic_as_str(&self) -> &'static str {
+        match NodeBasic::from_u8(self.get_basic()) {
+            Some(NodeBasic::Controller) => "Controller",
+            Some(NodeBasic::StaticController) => "Static Controller",
+            Some(NodeBasic::Slave) => "Slave",
+            Some(NodeBasic::RoutingSlave) => "Routing Slave",
+            _ => "unknown"
+        }
+    }
```

Comment:
nit: I don't know if this works, but worth trying implementing the Display trait for the NodeBasic enum ? (I don't know if it's possible at all)